tools/cli/internal/openapi/filter/info.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package filter
+
+import (
+	"fmt"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/mongodb/openapi/tools/cli/internal/apiversion"
+)
+
+type InfoFilter struct{}
+
+func (f *InfoFilter) Apply(oas *openapi3.T, metadata *Metadata) error {
+	return f.apply(oas.Info, metadata)
+}
+
+func (f *InfoFilter) apply(info *openapi3.Info, metadata *Metadata) error {
+	if info == nil {
+		return nil
+	}
+
+	if info.Description != "" {
+		info.Description = replaceVersion(info.Description, metadata.targetVersion)
+	}
+
+	return nil
+}
```

Comment:
why the struct if you don't reference `f`? and why the private method hidden behind the public method?

Also I think we need to revisit the linter it should be flagging `f` is unused

Reply:
you're right, we are initializing the filters without anything in the struct, I think we can initialize them with metadata, WDYT? I can do that in a follow-up PR

Reply:
@andreaangiolillo seem to be doing something similar on his PR so I'd ask you both to think on the design choices for th elib, do you need a state and as such a struct and how is that struct then initialized and used, and as "state" does that mean I can have multiple cuncurrently?

Reply:
yeah, we synced today and we think we can make use of keeping metadata in the filters, we initialize them with the right data since each filter should apply to a separate oas